scripts/Imaging/imgpy/helper.py

```diff
@@ -0,0 +1,340 @@
+
+    import time
+    print_string = ""
+    if _verbosity >= 2:
+        _timer_running = False
+        timer_string = str(time.time() - _timer_start)
+        print_string += (
+            str(message) + " Elapsed time: " + timer_string + " sec.")
+
+    if _verbosity >= 3:
+        print_string += " Memory usage after execution: " + get_memory_usage_linux(
+        )
+
+    tomo_print(_timer_print_prefix + print_string, verbosity)
+
+
+def total_execution_timer(message="Total execution time was "):
+    """
+    This will ONLY be used to time the WHOLE execution time.
+    The first call to this will be in tomo_reconstruct.py and it will start it.abs
+    The last call will be at the end of find_center or do_recon.
+    """
+    import time
+    global _whole_exec_timer
+    global _readme
+
+    if not _whole_exec_timer:
+        # change type from bool to timer
+        _whole_exec_timer = time.time()
+    else:
+        # change from timer to string
+        _whole_exec_timer = str(time.time() - _whole_exec_timer)
+        message += _whole_exec_timer + " sec"
+        if _readme is not None:
+            _readme.append(message)
+        print(message)
+
+
+def prog_init(total, desc="Progress", ascii=False, unit='images'):
+    """
+    Initialises and returns the progress bar if the tqdm library is available.
+
+    Otherwise does nothing, and Helper's progress update and close function will also do nothing.
+
+    :param total: the total number of iterations of the progress bar
+    :param desc: the label in front of the progress bar
+    :param ascii: to use ascii # (True) or utf-8 blocks (False)
+    :param unit: the unit for loading. Default is 'images'
+    """
+    global _progress_bar
+    if _verbosity > 0:
+        try:
+            from tqdm import tqdm
+            if _progress_bar is not None:
+                raise ValueError(
+                    "Timer was not closed previously. Please do prog_close()!")
+            _progress_bar = tqdm(
+                total=total, desc=desc, ascii=ascii, unit=unit)
+        except ImportError:
+            try:
+                from custom_timer import CustomTimer
+                _progress_bar = CustomTimer(total, desc)
+            except ImportError:
+                _progress_bar = None
+
+
+def prog_update(value=1):
+    """
+    This function will print a simple ascii bar if tqdm is not present.
+    """
+    global _progress_bar
+    if _progress_bar is not None:
+        _progress_bar.update(value)
+
+
+def prog_close():
+    """
+    This function will do nothing if the tqdm library is not present.
+    """
+    global _progress_bar
+    if _progress_bar is not None:
+        _progress_bar.close()
+
+    _progress_bar = None
+
+
+def set_readme(readme):
+    global _readme
+    _readme = readme
+
+
+def save_debug(sample, config, flat, dark, path_append, *args):
+    return
+    # if any of the arguments are none
+    for a in args:
+        if a is None or a is False:
```

Comment:
I've added a comment why we can't do `if not a`, the reason is that `a` might be an `numpy.ndarray` which will fail the check